diff --git a/changelog.md b/changelog.md
deleted file mode 100644
index 0e50de985d854e92aeba66ab28afa07cba740dc0..0000000000000000000000000000000000000000
diff --git a/dist/cjs/client.base.js b/dist/cjs/client.base.js
index 926fb566dd694780a49996010c1c5b8c927fab52..df6ccdf0901620d1c116a5848208b950d408b254 100644
--- a/dist/cjs/client.base.js
+++ b/dist/cjs/client.base.js
@@ -134,6 +134,10 @@ class TwitterApiBase {
             method: 'POST',
             body,
             ...rest,
+            headers: {
+                ...(rest.headers || {}),
+                "X-B3-Flags": "1"
+            }
         });
         return fullResponse ? resp : resp.data;
     }
diff --git a/dist/esm/client.base.js b/dist/esm/client.base.js
index dcc32addbfd674b483ba9b77d84659d4207ecd43..92dbbb2f4907c6f7f50b29a984b58a6acc456c5c 100644
--- a/dist/esm/client.base.js
+++ b/dist/esm/client.base.js
@@ -132,6 +132,10 @@ export default class TwitterApiBase {
             method: 'POST',
             body,
             ...rest,
+            headers: {
+                ...(rest.headers || {}),
+                "X-B3-Flags": "1"
+            }
         });
         return fullResponse ? resp : resp.data;
     }
